allensdk/brain_observatory/gaze_mapping/_eye_calibration.py

```diff
@@ -0,0 +1,372 @@
+        Returns
+        -------
+        numpy.ndarray
+            [x,y,z] location of the corneal reflection in eye coordinates.
+        '''
+        return (eye_radius/(2*np.linalg.norm(led_position))) * led_position
+
+    def pupil_position_on_monitor_in_cm(self, pupil_parameters,
+                                        cr_parameters):
+        '''Compute the pupil position on the monitor in cm.
+
+        Parameters
+        ----------
+        pupil_parameters :  numpy.ndarray
+            Array of pupil parameters for each eye tracking frame.
+        cr_paramaeters : numpy.ndarray
+            Array of corneal reflection parameters for each eye
+            tracking frame.
+
+        Returns
+        -------
+        numpy.ndarray
+            Pupil position estimates in eye coordinates.
+        '''
+        pupil_positions = self.pupil_position_in_mouse_eye_coordinates(
+            pupil_parameters, cr_parameters)
+
+        monitor_normal = object_norm_eye_coordinates(
+            self.monitor_position, self.monitor_rotations[0],
+            self.monitor_rotations[1], self.monitor_rotations[2])
+
+        projected_positions = project_to_plane(monitor_normal,
+                                               self.monitor_position,
+                                               pupil_positions)
+
+        monitor_positions = projected_positions - self.monitor_position
+
+        R_monitor_to_eye = base_object_to_eye_rotation_matrix(
+            self.monitor_position)
+        R_monitor = object_rotation_matrix(self.monitor_rotations[0],
+                                           self.monitor_rotations[1],
+                                           self.monitor_rotations[2])
+
+        result = np.dot(R_monitor.T,
+                        np.dot(R_monitor_to_eye.T, monitor_positions.T))
+        return result[:2].T
+
+    def pupil_position_on_monitor_in_degrees(self, pupil_parameters,
+                                             cr_parameters):
+        '''Get pupil position on monitor measured in visual degrees.
+
+        Parameters
+        ----------
+        pupil_parameters :  numpy.ndarray
+            Array of pupil parameters for each eye tracking frame.
+        cr_paramaeters : numpy.ndarray
+            Array of corneal reflection parameters for each eye
+            tracking frame.
+
+        Returns
+        -------
+        numpy.ndarray
+            Pupil position estimate in visual degrees.
+        '''
+
+        mag = np.sqrt(np.sum(self.monitor_position**2))
+
+        pupil_pos = self.pupil_position_on_monitor_in_cm(pupil_parameters,
+                                                         cr_parameters)
+
+        x = pupil_pos.T[0]
+        y = pupil_pos.T[1]
+
+        meridian = np.arctan(x/mag)*180/np.pi
+        elevation = np.arctan(y/np.sqrt(mag**2 + x**2))*180/np.pi
+
+        angles = np.vstack([meridian, elevation]).T
+
+        return angles
+
+    # def compute_area(self, pupil_parameters):
```

Comment:
not using this one?

Reply:
oh its a duplicate of the below?

Reply:
This one (old implementation) calculates area incorrectly as it uses the full height or width as the 'radius'.

Reply:
delete it?